addon/utils.js

```diff
@@ -440,20 +440,26 @@ export function removeInternalValues (val) {
 
   if (!Array.isArray(val)) {
     let withoutInternal
+    let merge
+    let without
     if (immutable.isImmutable(val)) {
-      withoutInternal = val.without('_internal')
-      const childrenWithoutInternals = removeChildInternals(withoutInternal)
-      return withoutInternal.merge(childrenWithoutInternals)
-    }
-
-    if (val._internal !== undefined) {
-      withoutInternal = Object.assign({}, val)
-      delete withoutInternal._internal
+      merge = immutable.merge
+      without = immutable.without
+    } else if (val._internal !== undefined) {
+      merge = Object.assign
+      without = _.omit
     } else {
-      withoutInternal = val
+      merge = function (first, second) {
+        if (Object.keys(second).length > 0) {
+          Object.assign({}, first, second) // don't mutate the object
```

Comment:
Also, what if the first was bigger then the second? Wouldn't you want to merge it still?